pageserver/src/repository.rs

```diff
@@ -69,6 +71,12 @@ pub trait Timeline {
     /// Truncate relation
     fn put_truncation(&self, rel: RelTag, lsn: Lsn, nblocks: u32) -> Result<()>;
 
+    /// Put raw data
+    fn put_raw_data(&self, key: RepositoryKey, data: &[u8]) -> Result<()>;
+
+    /// Get repository iterator
+    fn iterator(&self) -> Box<dyn RepositoryIterator + '_>;
```

Comment:
What order does it return the keys in? Does it return *all* data in the repository? Surely you don't want to scan the whole thing in most cases?

This seems pretty tightly coupled with the rocksdb implementation. I'm thinking that we would store each relation in a separate file, and with that it becomes pretty cumbersome to string them together into a single iterator.

Reply:
In rust I would expect this to be called `iter` or `into_iter`. It would be nice to return a concrete type, but I'm not sure if that's possible.

(Note `into_iter` is usually part of the `IntoIterator` trait.)